core/src/main/java/io/questdb/Bootstrap.java

```diff
@@ -188,14 +188,20 @@ public CairoConfiguration getCairoConfiguration() {
             log.errorW().$(e).$();
             throw new BootstrapException(e);
         }
-
-        // metrics
         if (config.getMetricsConfiguration().isEnabled()) {
             metrics = Metrics.enabled();
         } else {
             metrics = Metrics.disabled();
             log.advisoryW().$("Metrics are disabled, health check endpoint will not consider unhandled errors").$();
         }
+        Unsafe.setWriterMemLimit(config.getCairoConfiguration().getWriterMemoryLimit());
+        long rssMemoryLimit = config.getCairoConfiguration().getRssMemoryLimit();
+        if (rssMemoryLimit > 0) {
+            log.advisoryW().$(String.format("Setting RSS memory limit to %,d bytes", rssMemoryLimit)).$();
+            Unsafe.setRssMemLimit(rssMemoryLimit);
+        } else {
+            log.errorW().$("RSS memory limit is neither configured nor could be determined automatically").$();
```

Comment:
Why do we print an error message in case when RSS limit isn't configured? To me, this should be info or advisory logging.

Reply:
It is unusual that an automatic default couldn't be determined, and leaving the system with no RSS limit exposes it to a serious problem (OOM kill).

Reply:
Setting the prop to 0 enables the automatic limit. I guess they'd have to set it to a very large value in order to have "no limit".

Reply:
> Setting the prop to 0 enables the automatic limit.

We only call `detectRssLimitDefault()` as the default prop value, so if user sets `0` in `server.conf`, we end up with `0` as the RSS limit. When does automatic limit happen in this case?

Reply:
True, this was the earlier logic before this last change. Now if the user sets `0`, it disables the memory limit. I must correct the comment in the config file.